bootstrap.sh

```diff
@@ -0,0 +1,204 @@
+#!/bin/sh
+
+#
+# Bootstrap 0.1.0 for reMarkable (rM2 untested)
+# wget -qO- '<URL_to_bootstrap.sh>' | sh -s [packages]...
+#
+
+bootstrap_wget() {
+  if [ ! -d /home/root/.wget_bin ]
+  then
+      log INFO "Installing a current wget binary with better and current tls support" \
+          "to /home/root/.wget_bin/ for this installation..." \
+          ""
+      # Bootstrap a current version of wget
+      WGET_BINARIES_PATH='http://static.cosmos-ink.net/remarkable/artifacts'
+      WGET_BINARIES_FILENAME='wget-remarkable-pipeline_job245_wget1.20.3.zip'
+      WGET_BINARIES_SHA256='84185a5934e34e25794d439c78dc9f1590e4df12fbf369236f6a8749bf14d67f'
+
+      # Download and compare to hash
+      wget "$WGET_BINARIES_PATH/$WGET_BINARIES_FILENAME" -O "/home/root/$WGET_BINARIES_FILENAME"
+      if ! echo "$WGET_BINARIES_SHA256  /home/root/$WGET_BINARIES_FILENAME" | sha256sum -c -
+      then
+          echo "FATAL: Invalid hash" >&2
+          exit 1
+      fi
+
+      # Ensure to /home/root/.wget_bin exists and is empty
+      if [ -d /home/root/.wget_bin ]
+      then
+          rm -rf /home/root/.wget_bin/*
+      else
+          mkdir /home/root/.wget_bin
+      fi
+      # Unzip to /home/root/.wget_bin and remove downloaded file
+      unzip "/home/root/$WGET_BINARIES_FILENAME" -d /home/root/.wget_bin -q
+      rm "/home/root/$WGET_BINARIES_FILENAME"
+
+      cat > /home/root/.wget_bin/wget <<EOF
+#!/bin/sh
+LD_LIBRARY_PATH="/home/root/.wget_bin/dist" /home/root/.wget_bin/dist/wget \$@
+EOF
+
+      chmod +x /home/root/.wget_bin/wget
+  fi
+
+  # Ensure this binary is used
+  if [ `which wget` != '/home/root/.wget_bin/wget' ]; then
+    PATH="/home/root/.wget_bin:$PATH"
+  fi
+}
+
+
+main() {
+  # Select source for remarkable_entware repo
+  if [ -z "$REMARKABLE_ENTWARE_REPO_AUTHOR" ]; then
+    # Use LinusCDE's fork for the time beeing
+    # TODO: Switch to "Evidlo" as soon as PRs are merged
+    REMARKABLE_ENTWARE_REPO_AUTHOR=LinusCDE
+  fi
+
+
+  # Verify internet connection
+  if ! ping -c1 1.1.1.1 >/dev/null; then
+    throw "Couldn't connect to the internet. Please ensure that your reMarkable is connected to a WiFi-Network with internet access."
+  fi
+
+
+  # Verify dns
+  if ! ping -c1 remarkable.com >/dev/null; then
```

Comment:
no need to ping remarkable.com, just giving away info to them. maybe hit some other server

Reply:
Is there a good server you'd recommend? I took remarkable.com since it should never be down and everyone would probably trust it. If I'd just taken google.com, some people probably wouldn't have liked that.

Maybe github.com?

Reply:
Maybe toltec.delab.re or one.one.one.one?

Reply:
Cloudflare (one.one.one.one) would also be good. I don't want to take toltec.delab.re as this might at some point misreport no internet when there are in fact just maintenance works or similar. The big servers typically are all good for that since they're practicially never down at any given point.

We could also make the point, that I could discard those checks, since in order to run that script, most would download it with wget anyway.

Reply:
I removed those checks. Wouldn't matter if a one-liner tries to download that script in the first place.